presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1974,9 +1975,21 @@ public void createView(ConnectorSession session, ConnectorTableMetadata viewMeta
                 .put(PRESTO_QUERY_ID_NAME, session.getQueryId())
                 .build();
 
-        List<Column> columns = viewMetadata.getColumns().stream()
-                .map(col -> new Column(col.getName(), toHiveType(typeTranslator, col.getType()), Optional.ofNullable(col.getComment())))
-                .collect(toImmutableList());
+        List<Column> columns;
+        try {
+            columns = viewMetadata.getColumns().stream()
+                    .map(col -> new Column(col.getName(), toHiveType(typeTranslator, col.getType()), Optional.ofNullable(col.getComment())))
+                    .collect(toImmutableList());
+        }
+        catch (PrestoException e) {
+            // if a view uses any unsupported hive types, include only a dummy column value
+            if (e.getErrorCode().equals(NOT_SUPPORTED.toErrorCode())) {
+                columns = ImmutableList.of(new Column("dummy", HIVE_STRING, Optional.empty()));
+            }
```

Comment:
we should add a comment explaining why we have the `dummy` instead of proper column metadata. In my opinion, the message should be very clear - `Using dummy since column "abc" has an unsupported Hive type "xyz"` to avoid getting questions around why is my column metadata missing.